litgpt/deploy/serve.py

```diff
@@ -106,6 +123,42 @@ def encode_response(self, output: torch.Tensor) -> Dict[str, Any]:
         return {"output": decoded_output}
 
 
+class StreamLitAPI(BaseLitAPI):
+    def __init__(self,
+                 checkpoint_dir: Path,
+                 precision: Optional[str] = None,
+                 temperature: float = 0.8,
+                 top_k: int = 50,
+                 top_p: float = 1.0,
+                 max_new_tokens: int = 50):
+        super().__init__(checkpoint_dir, precision, temperature, top_k, top_p, max_new_tokens)   
+
+    def setup(self, device: str):
+        super().setup(device)
+
+    def predict(self, inputs: torch.Tensor) -> Any:
+        # Run the model on the input and return the output.
+        prompt_length = inputs.size(0)
+        max_returned_tokens = prompt_length + self.max_new_tokens
+
+        for block in self.model.transformer.h:
+            block.attn.kv_cache.reset_parameters()
+
+        yield generate(
+                self.model,
+                inputs,
+                max_returned_tokens,
+                temperature=self.temperature,
+                top_k=self.top_k,
+                top_p=self.top_p,
+                eos_id=self.tokenizer.eos_id
+                )
+
+    def encode_response(self, output_stream):
+        for outputs in output_stream:
+            yield [json.dumps({"output": self.tokenizer.decode(output)}) for output in outputs]
```

Comment:
LitServe can now do the https://github.com/Lightning-AI/LitServe/issues/84 so we can remove it from here.

Reply:
Oh good that you bring this up ... I remember now that I pinned the LitServe version, which maybe explains the need for this 😅
https://github.com/Lightning-AI/litgpt/blob/5dabf5f6be507e4da4d7be69813e6f31e799aefd/pyproject.toml#L41

Reply:
I think this has not been in the LitServe release yet right? So we would need to revisit this later in LitGPT.